web/controllers/certificateController/index.ts

```diff
@@ -118,92 +135,159 @@ export async function certificateHandler(req: Request, res: Response) {
  * @apiUse StatusInternalServerError
  */
 export async function confirmCertificateHandler(req: Request, res: Response) {
-    let status;
-    if (req.params.certificateId != undefined || ("" + req.params.certificateId)) {
-        return res.send(await viewParticipationCertificate(req.params.certificateId));
+    try {
+        const { certificateId } = req.params;
+        const entityManager = getManager();
+
+        if (typeof certificateId !== "string")
+            return res.status(400).send("Missing parameter certificateId");
+
+        const certificate = await entityManager.findOne(ParticipationCertificate, { uuid: certificateId.toUpperCase() }, { relations: ["student", "pupil"] });
+
+        if (!certificate)
+            return res.status(404).send("<h1>Zertifikatslink nicht valide.</h1>");
```

Comment:
It is discussable whether an invalid certificate should lead to a 404 here. Since the use case implies somehow the possible existence of invalid certificates, it is thus no technical error.

Reply:
I think 404 is fine here... I think it doesn't matter from an enduser's perspective who just opens the link through her browser.